frontend/src/features/properties/components/forms/ParcelDetailForm.tsx

```diff
@@ -270,26 +270,22 @@ const ParcelDetailForm = (props: ParcelPropertyProps) => {
           validate={handleValidate}
           enableReinitialize={true}
           onSubmit={async (values, actions) => {
-            let response: any;
             const apiValues = valuesToApiFormat(_.cloneDeep(values));
-            if (!values.id) {
-              response = dispatch(createParcel(apiValues));
-            } else {
-              response = dispatch(updateParcel(apiValues));
-            }
-            response
-              .then(() => {
-                history.push('/');
-              })
-              .catch((error: any) => {
-                const msg: string = error?.response?.data?.error ?? error.toString();
-                actions.setStatus({ msg });
-              })
-              .finally(() => {
-                actions.setSubmitting(false);
+            try {
+              if (!values.id) {
+                const data = await createParcel(apiValues)(dispatch);
                 dispatch(clear(actionTypes.ADD_PARCEL));
-                dispatch(clear(actionTypes.UPDATE_PARCEL));
-              });
+                history.replace(`/submitProperty/${data.id}`);
+              } else {
+                await updateParcel(apiValues)(dispatch);
+                window.location.reload();
```

Comment:
Not a big fan of this. This will force everything in the system to reload, redux, keycloak, all of the code values, etc. If we are going to do some kind of networking action, we should always use react router since this is a SPA.

I think your intention here is to forcibly get the page to show whatever values were saved to the DB? For that you could use history.go(0) to re-render the current page.

Reply:
I'd agree, we already have too many page refreshes for a SPA